package com.example.UserService.advice;

import java.sql.SQLIntegrityConstraintViolationException;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import com.example.UserService.Exception.ResourceNotFoundException;
import com.example.UserService.Payload.ResponseVo;

@RestControllerAdvice
public class userExceptionHandling {
	private Logger log = LoggerFactory.getLogger(this.getClass());
	
	@ResponseStatus(code = HttpStatus.NOT_FOUND)
	@ExceptionHandler(ResourceNotFoundException.class)
	public ResponseVo dataNotFoundException(ResourceNotFoundException exception) {
		log.error(exception.getMessage());

		return new ResponseVo(500, "Information not found enter valid user id..", "");
	}
	@ResponseStatus(code = HttpStatus.INTERNAL_SERVER_ERROR)
	@ExceptionHandler(SQLIntegrityConstraintViolationException.class)
	public ResponseVo duplicateEntryException(SQLIntegrityConstraintViolationException exception) {
		log.error(exception.getMessage());

		return new ResponseVo(500, "There is error to save save data in database", exception.getMessage());
	}
}
